### ISSUE TITLE
Investigate reason for file size increase in COLRv1 since v0.8.3

### ISSUE THREAD

**Issue by anthrotype** [googlefonts/nanoemoji#316]
cont'd from https://github.com/googlefonts/nanoemoji/issues/312#issuecomment-894177754

> we are seeing an increase in the overall length of the COLR table as a whole -- from 1,054,030 in the original [noto-glyf_colr_1.ttf] font to 1,364,730 in the new one (+310,700 bytes)

In particular, I see that the COLR.LayerList table size increased from 1,010,540 to 1,294,201 (+283,661) bytes. The total number of layers in the LayerList increased from 62782 to 73814 (+11,032).

**Comment by anthrotype** [comment]
Interestingly, the `font["maxp"].numGlyphs` (which counts all the glyphs in the font, including those used as PaintGlyph layers) _decreased_ from 45,459 to 41,793 (-3,666).

On the other hand, the total number of base color glyphs (`COLR.BaseGlyphList.BaseGlyphCount`) is the same in both old and new font: 3,573 (in fact the source files are the same for both).

**Comment by anthrotype** [comment]
I counted the number of unique Paint tables in the old and new font, both total and per-format.

For the old font we have a total of 81,151 paints, subdivided as:

```
{<PaintFormat.PaintGlyph: 10>: 55198,
 <PaintFormat.PaintLinearGradient: 4>: 4939,
 <PaintFormat.PaintColrLayers: 1>: 5828,
 <PaintFormat.PaintSolid: 2>: 3604,
 <PaintFormat.PaintTransform: 12>: 5307,
 <PaintFormat.PaintRadialGradient: 6>: 6275}
```

For the new font, we have a total of 96,063 (**+14,912**) subdivided as:
```
{<PaintFormat.PaintGlyph: 10>: 52497,
 <PaintFormat.PaintLinearGradient: 4>: 4939,
 <PaintFormat.PaintColrLayers: 1>: 6237,
 <PaintFormat.PaintSolid: 2>: 3604,
 <PaintFormat.PaintTransform: 12>: 21048,
 <PaintFormat.PaintComposite: 32>: 200,
 <PaintFormat.PaintTranslate: 14>: 248,
 <PaintFormat.PaintRadialGradient: 6>: 6275,
 <PaintFormat.PaintScale: 16>: 1015}
```

Most of the additional paints are PaintTransforms. Something to do with the new way in which nanoemoji implemented shape reuse (cf. https://github.com/googlefonts/nanoemoji/issues/302)

**Comment by rsheeter** [comment]
Fewer glyf and more Paint is expected for the updated implementation. We are probably now under-leveraging glyf composition. PaintTransform is fairly expensive; +15k of those isn't doing us any favors.

**Comment by rsheeter** [comment]
From a quick and dirty examination (util to examine in #320) of a relatively recent Noto build I see the following paint breakdown:

```
$ python tools/about_paint.py ~/oss/color-fonts/fonts/noto-glyf_colr_1.ttf
PaintColrLayers 6237
PaintComposite 200
PaintGlyph 52497
PaintLinearGradient 4939
PaintRadialGradient 6275
PaintScale 1015
PaintSolid 3604
PaintTransform 21048
PaintTransform::large_scale 173
PaintTransform::scale_around 11785
PaintTransform::scale_origin 390
PaintTranslate 248
PaintTransforms that should be upgraded have 4855 unique Affine2x3.
```

PaintTransform::something is giving detail of that paint transform. So, 173 are left as PaintTransform due to scales that don't fit in f2dot14, about 12k could convert to scales (#319 improves use of scale), and the remainder must set b,c and thus be some sort of rotation or skew.

Napkin math ignoring common format, offset<paint>:

* Before: PaintTransform is 7 bytes + 24 per unique 2x3, (11785+390) * 7 + 4855 * 24 = 201745 bytes
* After: PaintScaleAroundCenter is 12 bytes, (11785+390) * 12 = 146100
* Delta 55645

Interesting to see the impact of our specialized paints losing out on deduplication, it would be nice to have an offset to (sx, sy, cx, cy). Presumably there would be 4855 unique (same as the Affine2x3's we converted) so after would instead be (11785+390) * 7 + 4855 * 8 = 124065 or 22035 bytes smaller.

**Comment by rsheeter** [comment]
Per https://github.com/googlefonts/nanoemoji/pull/319#issuecomment-895433665, above overstates benefit as it doesn't currently check if the center is an int.
